e2e/cypress/integration/enterprise/group_mentions/group_mentions_permissions_spec.js

```diff
@@ -0,0 +1,386 @@
+                cy.apiDemoteUserToGuest(user.id);
+
+                // # Login as a guest user
+                cy.apiLogin(user);
+
+                // # Visit the channel
+                cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+
+                // # Type the Group Name
+                cy.get('#post_textbox').should('be.visible').clear().type(`@${groupName}`).wait(TIMEOUTS.TWO_SEC);
+
+                // * Verify if autocomplete dropdown is not displayed
+                cy.get('#suggestionList').should('not.be.visible');
+
+                // # Submit a post containing the group mention
+                cy.postMessage(`@${groupName}`);
+
+                // * Verify if a system message is not displayed
+                cy.getLastPostId().then((postId) => {
+                    cy.get(`#postMessageText_${postId}`).should('include.text', `@${groupName}`);
+
+                    // * Verify that the group mention is not highlighted
+                    cy.get(`#postMessageText_${postId}`).find('.mention--highlight').should('not.exist');
+
+                    // * Verify that the group mention does not has blue colored text
+                    cy.get(`#postMessageText_${postId}`).find('.group-mention-link').should('not.exist');
+                });
+
+                // # Login as sysadmin and enable group mentions permission for guests
+                cy.apiAdminLogin();
+                cy.visit('/admin_console/user_management/permissions/system_scheme');
+                enablePermission('guests-guest_use_group_mentions-checkbox');
+                saveConfig();
+
+                // # Login as guest user again and visit the channel
+                cy.apiLogin(user);
+                cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+
+                // # Type the Group Name
+                cy.get('#post_textbox').should('be.visible').clear().type(`@${groupName}`).wait(TIMEOUTS.TWO_SEC);
+
+                // * Verify if autocomplete dropdown is displayed
+                cy.get('#suggestionList').should('be.visible').children().within((el) => {
+                    cy.wrap(el).eq(0).should('contain', 'Group Mentions');
+                    cy.wrap(el).eq(1).should('contain', groupName);
+                });
+
+                // # Submit a post containing the group mention
+                cy.postMessage(`@${groupName}`);
+
+                // * Verify if a system message is displayed
+                cy.getLastPostId().then((postId) => {
+                    cy.get(`#postMessageText_${postId}`).should('include.text', `@${boardUser.username} did not get notified by this mention because they are not in the channel.`);
+
+                    // * Verify that the option to add them to channel is not given
+                    cy.get('a.PostBody_addChannelMemberLink').should('not.exist');
+                });
+            });
+        });
+    });
+
+    function enableGroupMention(groupName) {
```

Comment:
Just a preference I guess but could we move these functions to the top of the file rather than have them at the bottom? Non-blocking but I'll leave it up to your discretion on what should be done but just a personal coding preference.

Reply:
Thanks @hahmadia. But we decided to have these functions at the bottom to make it consistent. Some of the old files might still not have this updated, but at least for new ones we use this pattern. I will see if the documentation can reflect this. Thanks for pointing it out though.